lib/services/messenger/translators/discourse.ts

```diff
@@ -240,19 +240,29 @@ export class DiscourseTranslator extends TranslatorScaffold implements Translato
 	 * @returns         Promise that resolves to emit instructions.
 	 */
 	private static convertReadConnectionResponse(
-		message: TransmitInformation, response: DiscourseResponse
+		message: TransmitInformation, response: DiscoursePostSearchResponse
 	): Promise<IdentifyThreadResponse> {
-		const idFinder = new RegExp(`${message.source.service} thread ([\\w\\d-+\\/=]+)`, 'i');
-		if (response.posts.length > 0) {
-			return Promise.resolve({
+		const idFinder = new RegExp(`${message.source.service} thread ([\\w\\d-+\\/=\\_]+)`, 'i');
```

Comment:
This code seems like it could be generalised and shared between the two classes.

Reply:
If this is meant to be the same as the finder in `flowdock.ts`, then this def. needs commonality so errors can't creep in (like the difference that currently exists) as Lucian's pointed out.